core/src/main/java/io/confluent/kafka/schemaregistry/rest/RequestIdHandler.java

```diff
@@ -31,24 +31,33 @@
 
 public class RequestIdHandler extends HandlerWrapper {
   public static final String REQUEST_ID_HEADER = "X-Request-ID";
+  public static final String CALLER_IP_HEADER = "X-Forwarded-For";
 
   @Override
   public void handle(String target, Request baseRequest, HttpServletRequest request,
                      HttpServletResponse response) throws IOException, ServletException {
-    // Clear MDC at beginning of each request to remove stale values
+    // Clear MDC at the beginning of each request to remove stale values
     MDC.clear();
     MutableHttpServletRequest mutableRequest = new MutableHttpServletRequest(request);
-    List<String> inputHeaders = Collections.list(baseRequest.getHeaders(REQUEST_ID_HEADER));
+    addRequestIdToRequest(baseRequest, mutableRequest, response);
+    addCallerIpToRequest(mutableRequest, request);
+
+    // Call the next handler in the chain
+    super.handle(target, baseRequest, mutableRequest, response);
+  }
 
+  protected void addRequestIdToRequest(Request baseRequest, MutableHttpServletRequest mutableRequest, HttpServletResponse response) {
```

Comment:
its rather difficult to test mutableRequest portion without separating into funcs like this